sympy/core/relational.py

```diff
@@ -543,6 +543,49 @@ def _eval_simplify(self, ratio, measure, rational, inverse):
                 pass
         return e.canonical
 
+    def subs(self, *args, **kwargs):  # should mirror sympy.core.basic.subs
+        r"""Return a new equation with subs applied to both sides.
+
+        **Examples:**
+
+        >>> from sympy import Eq, symbols
+        >>> a, b, c, x, y, z = symbols('a b c x y z')
+        >>> eq1 = Eq(y, a ** 3 + b ** 2 + c)
+        >>> eq_a = Eq(a, x / 2)
+        >>> eq_b = Eq(b, 2 * x)
+        >>> eq_c = Eq(c, x + 2)
+        >>> eq1.subs({a: eq_a.rhs, b: eq_b.rhs, c: eq_c.rhs})
+        Eq(y, x**3/8 + 4*x**2 + x + 2)
+        >>> eq1.subs({a: eq_a.rhs, b: eq_b.rhs})
+        Eq(y, c + x**3/8 + 4*x**2)
+        >>> eq1.subs(a, eq_a.rhs)
+        Eq(y, b**2 + c + x**3/8)
+        >>> eq1.subs(eq_a, eq_b, eq_c)
+        Eq(y, x**3/8 + 4*x**2 + x + 2)
+        >>> eq1.subs(eq_a, eq_b)
+        Eq(y, c + x**3/8 + 4*x**2)
+        >>> eq1.subs(eq_a)
+        Eq(y, b**2 + c + x**3/8)        
+
+        """
+        sequence = args
```

Comment:
maybe just `return Basic.subs(self, [i if not isinstance(i, Eq) else i.args for i in sequence], **kwargs)`?

Reply:
Thanks for looking into this, but I am lost a bit. Are you suggesting to replace the whole block of code by these two lines? If this achieves the desired behaviour, that would be great!

Reply:
Actually, this works. Thank you, @smichr! I just updated it.